adapters/testutils/synctransaction_test_utils.go

```diff
@@ -64,12 +64,13 @@ func (b *SyncTransactionBuilder[C, P]) GetTestDeclareV0Transaction(
 		FeeDAMode:             0,   // this field is not available on v0
 	}
 
-	var p2pHash *common.Hash
-	consensusDeclareTransaction.TransactionHash, p2pHash = getTransactionHash(
+	transactionHash, p2pHash := getTransactionHash(
```

Comment:
```suggestion
	*consensusDeclareTransaction.TransactionHash, p2pHash := getTransactionHash(
```
Please also fix the other places, thanks.

Reply:
This requires me to initialize `consensusDeclareTransaction.TransactionHash` first, otherwise nil pointer deref. Would you prefer this path?